lib/event_tracer/basic_decorator.rb

```diff
@@ -16,5 +16,17 @@ def __setobj__(obj)
       @delegate_sd_obj = obj
     end
 
+    def valid_args?(metrics)
+      metrics && (metrics.is_a?(Hash) || metrics.is_a?(Array))
+    end
```

Comment:
Is this applicable for all loggers? If it doesn't, should we move it to individual logger instead?